sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -56,7 +56,7 @@ private object MsSqlServerDialect extends JdbcDialect {
   }
 
   override def getJDBCType(dt: DataType): Option[JdbcType] = dt match {
-    case TimestampType => Some(JdbcType("DATETIME", java.sql.Types.TIMESTAMP))
+    case TimestampType => Some(JdbcType("DATETIME2", java.sql.Types.TIMESTAMP))
```

Comment:
This may cause some problems on the existing Spark apps on MsSQLServer tables. What happens when you try to access the tables created by old Spark versions?

Reply:
The old Spark versions maps to datetime, which is compatible with datetime2. datatime2 will also allow more digits of milliseconds.